app/src/main/kotlin/com/flxrs/dankchat/preferences/userdisplay/UserDisplayDto.kt

```diff
@@ -0,0 +1,97 @@
+@file:OptIn(ExperimentalContracts::class)
+
+package com.flxrs.dankchat.preferences.userdisplay
+
+import androidx.annotation.ColorInt
+import com.flxrs.dankchat.data.database.UserDisplayEntity
+import kotlinx.serialization.Serializable
+import kotlin.contracts.ExperimentalContracts
+import kotlin.contracts.contract
+
+@Serializable
+data class UserDisplayDto(
```

Comment:
This file serves no real purpose at the moment. It is located in the ui package but is also part of some of the messages, which are located in the data package.
I would instead like the approach like I did for Highlights/Ignores. I have some kind of model that has only the properties needed for the messages, like the class `Highlight` this would be located in the different Message classes.
For the bottom sheet, the `UserDisplayViewModel` can directly consume the database entities and map them to the `UserDisplayItem`, skipping this dto mapping completely.

Reply:
the `Dto` class will be removed, and `UserDisplayViewModel` will be implemented :+1: